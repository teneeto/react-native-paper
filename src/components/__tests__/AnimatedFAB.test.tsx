/// <reference types="@testing-library/jest-native" />

import * as React from 'react';
import { Animated, StyleSheet } from 'react-native';

import { render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import { MD3Colors } from '../../styles/themes/v3/tokens';
import AnimatedFAB from '../FAB/AnimatedFAB';

const styles = StyleSheet.create({
  background: {
    backgroundColor: MD3Colors.tertiary50,
  },
});

it('renders animated fab', () => {
  const tree = renderer
    .create(
      <AnimatedFAB onPress={() => {}} label="" extended={false} icon="plus" />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders animated fab with label on the right by default', () => {
  const tree = renderer
    .create(
      <AnimatedFAB label="text" extended onPress={() => {}} icon="plus" />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders animated fab with label on the left', () => {
  const tree = renderer
    .create(
      <AnimatedFAB
        label="text"
        extended
        animateFrom="left"
        onPress={() => {}}
        icon="plus"
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders animated fab with only transparent container', () => {
  const { getByTestId } = render(
    <AnimatedFAB
      label="text"
      animateFrom="left"
      onPress={() => {}}
      icon="plus"
      testID="animated-fab"
      extended={false}
      style={styles.background}
    />
  );
  expect(getByTestId('animated-fab-container')).toHaveStyle({
    backgroundColor: 'transparent',
  });
});

it('animated value changes correctly', () => {
  const value = new Animated.Value(1);
  const { getByTestId } = render(
    <AnimatedFAB
      label="text"
      icon="plus"
      testID="animated-fab"
      extended={false}
      style={[{ transform: [{ scale: value }] }]}
    />
  );
  expect(getByTestId('animated-fab-container-outer-layer')).toHaveStyle({
    transform: [{ scale: 1 }],
  });

  Animated.timing(value, {
    toValue: 1.5,
    useNativeDriver: false,
    duration: 200,
  }).start();

  jest.advanceTimersByTime(200);

  expect(getByTestId('animated-fab-container-outer-layer')).toHaveStyle({
    transform: [{ scale: 1.5 }],
  });
});

it('renders FAB without uppercase styling if uppercase prop is falsy', () => {
  const { getByTestId } = render(
    <AnimatedFAB
      label="text"
      animateFrom="left"
      onPress={() => {}}
      icon="plus"
      testID="animated-fab"
      style={styles.background}
      extended={false}
      uppercase={false}
    />
  );

  expect(getByTestId('animated-fab-text')).not.toHaveStyle({
    textTransform: 'uppercase',
  });
});

it('renders FAB with uppercase styling if uppercase prop is truthy', () => {
  const { getByTestId } = render(
    <AnimatedFAB
      label="text"
      animateFrom="left"
      onPress={() => {}}
      icon="plus"
      testID="animated-fab"
      style={styles.background}
      extended={false}
      uppercase
    />
  );

  expect(getByTestId('animated-fab-text')).toHaveStyle({
    textTransform: 'uppercase',
  });
});
