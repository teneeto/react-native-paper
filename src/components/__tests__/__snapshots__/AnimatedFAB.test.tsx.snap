// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders animated fab 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignSelf": undefined,
      "backgroundColor": "transparent",
      "borderRadius": 16,
      "bottom": undefined,
      "end": undefined,
      "flex": undefined,
      "height": undefined,
      "left": undefined,
      "opacity": 1,
      "position": "absolute",
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 8,
      "start": undefined,
      "top": undefined,
      "transform": Array [
        Object {
          "scale": 1,
        },
      ],
      "width": undefined,
    }
  }
  testID="animated-fab-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderRadius": 16,
        "flex": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
    testID="animated-fab-container"
  >
    <View
      collapsable={false}
      style={
        Object {
          "borderRadius": 16,
          "height": 56,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "elevation": 0,
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            Object {
              "borderRadius": 16,
              "bottom": 0,
              "elevation": 6,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 72,
            }
          }
        />
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            Object {
              "borderRadius": 16,
              "bottom": 0,
              "elevation": 6,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 16,
                },
              ],
              "width": 56,
            }
          }
        />
      </View>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "borderRadius": 16,
            "flexDirection": "row",
            "overflow": "hidden",
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "rgba(234, 221, 255, 1)",
              "borderRadius": 16,
              "height": 56,
              "left": 16,
              "right": undefined,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": 72,
            }
          }
        >
          <View
            accessibilityLabel=""
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                false,
                Object {
                  "borderRadius": 16,
                },
              ]
            }
            testID="animated-fab"
          >
            <View
              style={
                Array [
                  Object {
                    "height": 56,
                  },
                  Object {
                    "borderRadius": 16,
                    "width": 72,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        Object {
          "alignItems": "center",
          "height": 56,
          "justifyContent": "center",
          "left": 16,
          "position": "absolute",
          "right": undefined,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
          "width": 56,
        }
      }
    >
      <Text
        accessibilityElementsHidden={true}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        selectable={false}
        style={
          Array [
            Object {
              "color": "rgba(33, 0, 93, 1)",
              "fontSize": 24,
            },
            Array [
              Object {
                "lineHeight": 24,
                "transform": Array [
                  Object {
                    "scaleX": 1,
                  },
                ],
              },
              Object {
                "backgroundColor": "transparent",
              },
            ],
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        Û∞êï
      </Text>
    </View>
    <View
      pointerEvents="none"
    >
      <Text
        collapsable={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        onTextLayout={[Function]}
        style={
          Object {
            "color": "rgba(33, 0, 93, 1)",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "500",
            "letterSpacing": 0.1,
            "lineHeight": 20,
            "minWidth": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 16,
            "textAlign": "left",
            "top": -28,
            "transform": Array [
              Object {
                "translateX": 56,
              },
            ],
            "writingDirection": "ltr",
          }
        }
        testID="animated-fab-text"
      />
    </View>
  </View>
</View>
`;

exports[`renders animated fab with label on the left 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignSelf": undefined,
      "backgroundColor": "transparent",
      "borderRadius": 16,
      "bottom": undefined,
      "end": undefined,
      "flex": undefined,
      "height": undefined,
      "left": undefined,
      "opacity": 1,
      "position": "absolute",
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 8,
      "start": undefined,
      "top": undefined,
      "transform": Array [
        Object {
          "scale": 1,
        },
      ],
      "width": undefined,
    }
  }
  testID="animated-fab-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderRadius": 16,
        "flex": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
    testID="animated-fab-container"
  >
    <View
      collapsable={false}
      style={
        Object {
          "borderRadius": 16,
          "height": 56,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "elevation": 0,
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            Object {
              "borderRadius": 16,
              "bottom": 0,
              "elevation": 6,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 72,
            }
          }
        />
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            Object {
              "borderRadius": 16,
              "bottom": 0,
              "elevation": 6,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": 56,
            }
          }
        />
      </View>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "borderRadius": 16,
            "flexDirection": "row",
            "overflow": "hidden",
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "rgba(234, 221, 255, 1)",
              "borderRadius": 16,
              "height": 56,
              "left": -16,
              "right": undefined,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": 72,
            }
          }
        >
          <View
            accessibilityLabel="text"
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                false,
                Object {
                  "borderRadius": 16,
                },
              ]
            }
            testID="animated-fab"
          >
            <View
              style={
                Array [
                  Object {
                    "height": 56,
                  },
                  Object {
                    "borderRadius": 16,
                    "width": 72,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        Object {
          "alignItems": "center",
          "height": 56,
          "justifyContent": "center",
          "left": -16,
          "position": "absolute",
          "right": undefined,
          "transform": Array [
            Object {
              "translateX": 16,
            },
          ],
          "width": 56,
        }
      }
    >
      <Text
        accessibilityElementsHidden={true}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        selectable={false}
        style={
          Array [
            Object {
              "color": "rgba(33, 0, 93, 1)",
              "fontSize": 24,
            },
            Array [
              Object {
                "lineHeight": 24,
                "transform": Array [
                  Object {
                    "scaleX": 1,
                  },
                ],
              },
              Object {
                "backgroundColor": "transparent",
              },
            ],
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        Û∞êï
      </Text>
    </View>
    <View
      pointerEvents="none"
    >
      <Text
        collapsable={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        onTextLayout={[Function]}
        style={
          Object {
            "color": "rgba(33, 0, 93, 1)",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "500",
            "left": 16,
            "letterSpacing": 0.1,
            "lineHeight": 20,
            "minWidth": 0,
            "opacity": 0,
            "position": "absolute",
            "textAlign": "left",
            "top": -28,
            "transform": Array [
              Object {
                "translateX": 56,
              },
            ],
            "writingDirection": "ltr",
          }
        }
        testID="animated-fab-text"
      >
        text
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders animated fab with label on the right by default 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignSelf": undefined,
      "backgroundColor": "transparent",
      "borderRadius": 16,
      "bottom": undefined,
      "end": undefined,
      "flex": undefined,
      "height": undefined,
      "left": undefined,
      "opacity": 1,
      "position": "absolute",
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 8,
      "start": undefined,
      "top": undefined,
      "transform": Array [
        Object {
          "scale": 1,
        },
      ],
      "width": undefined,
    }
  }
  testID="animated-fab-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderRadius": 16,
        "flex": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
    testID="animated-fab-container"
  >
    <View
      collapsable={false}
      style={
        Object {
          "borderRadius": 16,
          "height": 56,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "elevation": 0,
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            Object {
              "borderRadius": 16,
              "bottom": 0,
              "elevation": 6,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 72,
            }
          }
        />
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            Object {
              "borderRadius": 16,
              "bottom": 0,
              "elevation": 6,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 16,
                },
              ],
              "width": 56,
            }
          }
        />
      </View>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "borderRadius": 16,
            "flexDirection": "row",
            "overflow": "hidden",
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "rgba(234, 221, 255, 1)",
              "borderRadius": 16,
              "height": 56,
              "left": 16,
              "right": undefined,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": 72,
            }
          }
        >
          <View
            accessibilityLabel="text"
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                false,
                Object {
                  "borderRadius": 16,
                },
              ]
            }
            testID="animated-fab"
          >
            <View
              style={
                Array [
                  Object {
                    "height": 56,
                  },
                  Object {
                    "borderRadius": 16,
                    "width": 72,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        Object {
          "alignItems": "center",
          "height": 56,
          "justifyContent": "center",
          "left": 16,
          "position": "absolute",
          "right": undefined,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
          "width": 56,
        }
      }
    >
      <Text
        accessibilityElementsHidden={true}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        selectable={false}
        style={
          Array [
            Object {
              "color": "rgba(33, 0, 93, 1)",
              "fontSize": 24,
            },
            Array [
              Object {
                "lineHeight": 24,
                "transform": Array [
                  Object {
                    "scaleX": 1,
                  },
                ],
              },
              Object {
                "backgroundColor": "transparent",
              },
            ],
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        Û∞êï
      </Text>
    </View>
    <View
      pointerEvents="none"
    >
      <Text
        collapsable={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        onTextLayout={[Function]}
        style={
          Object {
            "color": "rgba(33, 0, 93, 1)",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "500",
            "letterSpacing": 0.1,
            "lineHeight": 20,
            "minWidth": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 16,
            "textAlign": "left",
            "top": -28,
            "transform": Array [
              Object {
                "translateX": 56,
              },
            ],
            "writingDirection": "ltr",
          }
        }
        testID="animated-fab-text"
      >
        text
      </Text>
    </View>
  </View>
</View>
`;
